src/cfamily/ondemand.ts

```diff
@@ -7,23 +7,32 @@
 'use strict';
 
 import * as fs from 'fs';
+import { DateTime } from 'luxon';
 import * as openpgp from 'openpgp';
 import * as path from 'path';
 import * as vscode from 'vscode';
 import * as util from '../util/util';
 
+// Comparing a `DateTime` in the past with `diffNow` returns a negative number
+const PLUGIN_MAX_AGE_MONTHS = -2;
+
 const CFAMILY_PLUGIN_ID = 'sonar-cfamily-plugin';
 const CFAMILY_JAR = 'sonarcfamily.jar';
 
+function getOnDemandAnalyzersPath() {
+  return path.resolve(util.extensionPath, '..', 'sonarsource.sonarlint_ondemand-analyzers');
+}
+
 export function maybeAddCFamilyJar(params: string[]) {
   const expectedVersion: string = util.packageJson.jarDependencies.filter(dep => dep.artifactId === CFAMILY_PLUGIN_ID)[0].version;
-  const onDemandAnalyzersPath = path.resolve(util.extensionPath, '..', 'sonarsource.sonarlint_ondemand-analyzers');
-  const maybeCFamilyJar = path.resolve(onDemandAnalyzersPath, CFAMILY_PLUGIN_ID, expectedVersion, CFAMILY_JAR);
+  const maybeCFamilyJar = path.resolve(getOnDemandAnalyzersPath(), CFAMILY_PLUGIN_ID, expectedVersion, CFAMILY_JAR);
   if (fs.existsSync(maybeCFamilyJar)) {
     params.push(maybeCFamilyJar);
+    util.extensionContext.globalState.update(lastUsedKey(CFAMILY_PLUGIN_ID, expectedVersion), DateTime.now().toMillis());
```

Comment:
I think this method returns Thenable? So maybe, in theory, it is possible to quickly remove the currently needed analyzer before the state is updated.

I would suggest to either await this update, or only iterate through the analyzers that are NOT of `expectedVersion` later on.